File: 27_이유리/1주차/햄버거만들기.java
import java.util.Stack;

class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack stack = new Stack();
        int sSize;
        
        for(int i : ingredient) {
        	stack.push(i);
        	sSize = stack.size();
        	
        	// 재료들이 쌓인 상태가 햄버거를 포장할 수 있는지 확인
        	if(sSize >= 4 && 
        			stack.elementAt(sSize-4).equals(1) && 
        			stack.elementAt(sSize - 3).equals(2) &&
        			stack.elementAt(sSize - 2).equals(3) && 
        			stack.elementAt(sSize - 1).equals(1)) {
        		answer++;					// 햄버거 포장이 가능하면 answer++
        		for(int j = 0; j < 4; j++)	// 햄버거를 포장하고 나면 pop으로 사용한 재료 삭제
        			stack.pop();
        	}
        }
        
        return answer;
    }
}

File: 21_박민혁/햄버거 만들기.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    int answer = 0;

    for (int i = 0; i < ingredient.size() - 3; i++) {
        if (ingredient[i] == 1) {
            if (ingredient[i + 1] == 2) {
                if (ingredient[i + 2] == 3) {
                    if (ingredient[i + 3] == 1) {
                        ingredient.erase(ingredient.begin() + i);
                        ingredient.erase(ingredient.begin() + i);
                        ingredient.erase(ingredient.begin() + i);
                        ingredient.erase(ingredient.begin() + i);

                        i = 0;
                        answer++;
                        //for (int j = 0; j < ingredient.size(); j++) {
                        //    cout << ingredient[j];
                        //}
                        //cout << endl;
                    }
                }
            }
        }
    }
    return answer;
}

int main() {
    vector<int> ingredient; //[2, 1, 1, 2, 3, 1, 2, 3, 1], [1, 3, 2, 1, 2, 1, 3, 1, 2]
    ingredient.push_back(1);
    ingredient.push_back(3);
    ingredient.push_back(2);
    ingredient.push_back(1);
    ingredient.push_back(2);
    ingredient.push_back(1);
    ingredient.push_back(3);
    ingredient.push_back(1);
    ingredient.push_back(2);
    int burger = solution(ingredient);
    cout << burger << endl;

    ingredient.push_back(2);
    ingredient.push_back(1);
    ingredient.push_back(1);
    ingredient.push_back(2);
    ingredient.push_back(3);
    ingredient.push_back(1);
    ingredient.push_back(2);
    ingredient.push_back(3);
    ingredient.push_back(1);
    burger = solution(ingredient);
    cout << burger << endl;

    return 0;
}
